src/main/java/org/broadinstitute/hdf5/HDF5File.java

```diff
@@ -477,21 +510,21 @@ private void checkIsOpen() {
     /**
      * Opens a HDF5 file given its access {@link OpenMode mode}.
      */
-    private static int open(final File file, final OpenMode mode) {
+    private static int open(final Path file, final OpenMode mode) {
         final int fileId;
         try {
             if (mode == OpenMode.CREATE) {
-                file.delete();
-                file.createNewFile();
+                Files.deleteIfExists(file);
+                Files.createFile(file);
             }
-            fileId = H5.H5Fopen(file.getAbsolutePath(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
+            fileId = H5.H5Fopen(file.toAbsolutePath().toString(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
```

Comment:
Do you mean to copy to a temp directory the HDFS/GCS file and read from there? In that case, how the `READ_WRITE` option should work? Two ideas about that, but with some problems for both:

1. Write only the temp file, and copy back to the "original" `Path` on close. This will not update the provided path until the HDF5File is closed, so that means that changes made by the API won't be reflected if the file is open with a second holder.
2. Every time that a dataset is added to the temp file, copy back to the "original" Path. This will have performance issues unless we can append just the generated data.

Waiting for your feedback on how to proceed.